### PR TITLE
Remove .99 when possible

### PR DESCRIPTION
**Pull request by haampie**
After https://github.com/spack/spack/pull/26402 we can drop almost all .999's

Actually many .999's were redundant even without that pr.

I've glanced over most changes and verified there's no `x:x`, `x.y:x.y`, `x.y.z:x.y.z` using these kinds of regexes `(\d+\.\d+):\1["', ]`

### CODE REVIEW

File: var/spack/repos/builtin/packages/petsc/package.py [hunk 1]
```diff
@@ -265,14 +265,14 @@ class Petsc(Package, CudaPackage, ROCmPackage):
...
-    depends_on('superlu-dist@5.0.0:5.1.3~int64', when='@3.7:3.7.99+superlu-dist+mpi~int64')
-    depends_on('superlu-dist@5.0.0:5.1.3+int64', when='@3.7:3.7.99+superlu-dist+mpi+int64')
-    depends_on('superlu-dist@5.2:5.2.99~int64', when='@3.8:3.9.99+superlu-dist+mpi~int64')
-    depends_on('superlu-dist@5.2:5.2.99+int64', when='@3.8:3.9.99+superlu-dist+mpi+int64')
-    depends_on('superlu-dist@5.4:5.4.99~int64', when='@3.10:3.10.2+superlu-dist+mpi~int64')
-    depends_on('superlu-dist@5.4:5.4.99+int64', when='@3.10:3.10.2+superlu-dist+mpi+int64')
-    depends_on('superlu-dist@6.1:6.1.99~int64', when='@3.10.3:3.12.99+superlu-dist+mpi~int64')
-    depends_on('superlu-dist@6.1:6.1.99+int64', when='@3.10.3:3.12.99+superlu-dist+mpi+int64')
+    depends_on('superlu-dist@5.0.0:5.1.3~int64', when='@3.7.0:3.7+superlu-dist+mpi~int64')
+    depends_on('superlu-dist@5.0.0:5.1.3+int64', when='@3.7.0:3.7+superlu-dist+mpi+int64')
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/petsc/package.py:269]: I don't think the introduction of `.0` here is necessary unless you think `N.0:N` is clearer than `N:N` (I personally think it's ungainly).

**Review by sethrj — COMMENTED** [review]
Github is having a little trouble with this many changed files, but I  just have a question about the .0 files. I think you can also expand the scope lightly to replace some of the old and new `N.0:N` constraints to `N:N` which I think looks better.

**Comment by haampie** [comment]
@sethrj unfortunately `x:x` is not supported by Spack right now :(. It's OK to create a `VersionRange(x, x)`, but Spack simplifies it to `Version(x)` in some code paths, which behaves in other code paths as a concrete version and elsehwere as a range. Fixing that bug requires some work...

**Comment by sethrj** [comment]
Yuck! OK, thanks for the clarification. My check of the changes was more than a spot check but less than reviewing every change. The E4S pipeline passes, and I'd say let's get this in here before more merge conflicts develop. And the `:N.99` is of course something we'll have to watch out for in future code reviews :)

**Review by sethrj — APPROVED** [review]

**Comment by sethrj** [comment]
Ahh right, the `x:x` problem is already in an open PR of yours 😄  https://github.com/spack/spack/pull/26208

**Pull request merged**